### ISSUE TITLE
Conversion issues when writing go signatures and input is json/gob

### ISSUE THREAD

**Issue by michaelkatch** [aquasecurity/tracee#865]
Hey,

I have encountered issues when writing Golang signatures those issues are always related to conversions from type "tracce argument" to the real type of argument. 
For example when dealing with "magic_write" event :
	      *bytes, err := helpers.GetTraceeArgumentByName(eventObj, "bytes")
		      if err != nil {
			      return err
		      }
		      bytesValue := bytes.Value.([]byte)*
![image](https://user-images.githubusercontent.com/70207874/128012331-93cb6e2c-09f8-4e53-80f0-03962a6ddfcc.png)

It will work when the input is gob but won't work when the input is JSON.
The same applies to arguments that need to be converted such as commit_creds struct and more.

Basically, I'm asking that the user won't be dependent on the type of input of "tracee-ebpf".
